src/game/MatrixGeneration.ts

```diff
@@ -0,0 +1,249 @@
+import GameConfiguration from "./GameConfiguration";
+
+import seedrandom from "seedrandom";
+
+//TODO should this defined here?
+//TODO how to choose these numbers?
+//TODO is this correct way to provide constants? (what are the naming conventions?)
+const MinMatrixSize = 3;
+const MaxMatrixSize = 8;
+
+const MinNumberSequences = 2;
+const MaxNumberSequences = 7;
+const MinSequenceLength = 2;
+
+const MinBufferSize = 4;
+const MaxBufferSize = 8;
+
+const StepsUntilMatrixSizeIncreased = 10;
+const StepsUntilSequenceNumberIncreased = 3;
+
+const DefaultTimeout = 60_000;
```

Comment:
We should define this in one place (duplication with GameState right now).

Reply:
I agree. What's the best place for this? @Ablu , what do you think?
What about moving all the constants to some place else? Maybe providing some way such that these values aren't hard coded would be nice.